index.js

```diff
@@ -40,10 +46,14 @@ exports.run = function(runner, specs) {
         } catch (err) {
           reject(err);
         }
-      });
+      }
     });
   });
 
+  function isCucumber2() {
+    return !!Cucumber.defineSupportCode;
```

Comment:
ahh, yes that makes sense but would be a rework also can we start using ES-6 features or would it cost/break existing version ?
* native promises instead of q library.
* `Object.assign()` method instead of using object-assign library etc.